lib/linguist/heuristics.yml

```
@@ -492,6 +492,13 @@ disambiguations:
   - language: NL
     pattern: '^(b|g)[0-9]+ '
   - language: NewLisp
+- extensions: ['.nr']
+  rules:
+  - language: Noir
+    and:
+    - negative_pattern: '^\.'
+    - pattern: '^\s*fn[ \t]+[^\(\s].*?\('
+  - language: Roff
```

This second pattern can match valid Roff markup too. For example, an author might markup a code sample using:

~~~roff
.TP Filename: rust.rs
.EX
	fn main() {
		println!("Hello, world!");
	}
.EE
~~~

A less problematic (as well as simpler) heuristic would be:

```suggestion
  - language: Roff
    pattern: '^\.'
  - language: Noir
```

… as a Roff document without any formatting commands is extremely rare in practice.

Also, I'm not even sure about the open-ended fallback to Noir, quite frankly, because if the Roff pattern doesn't match, Linguist falls through to the classifier strategy (the one that's trailed on our [`samples/`](https://github.com/github-linguist/linguist/tree/master/samples) directory). This is generally the preferred approach.